src/Razor/src/Microsoft.AspNetCore.Razor.ExternalAccess.LegacyEditor/RazorWrapperFactory.cs

```diff
@@ -0,0 +1,201 @@
+        => Wrap<RazorEditorFactoryService, EditorFactoryServiceWrapper, IRazorEditorFactoryService>(obj, static obj => new EditorFactoryServiceWrapper(obj));
+
+    internal static IRazorEditorSettingsManager WrapEditorSettingsManager(object obj)
+        => Wrap<EditorSettingsManager, EditorSettingsManagerWrapper, IRazorEditorSettingsManager>(obj, static obj => new EditorSettingsManagerWrapper(obj));
+
+    internal static IRazorElementCompletionContext WrapElementCompletionContext(object obj)
+        => Wrap<ElementCompletionContext, ElementCompletionContextWrapper, IRazorElementCompletionContext>(obj, static obj => new ElementCompletionContextWrapper(obj));
+
+    internal static IRazorParser WrapParser(object obj)
+        => Wrap<VisualStudioRazorParser, ParserWrapper, IRazorParser>(obj, static obj => new ParserWrapper(obj));
+
+    internal static IRazorRequiredAttributeDescriptor WrapRequiredAttributeDescriptor(object obj)
+        => Wrap<RequiredAttributeDescriptor, RequiredAttributeDescriptorWrapper, IRazorRequiredAttributeDescriptor>(obj, static obj => new RequiredAttributeDescriptorWrapper(obj));
+
+    internal static IRazorTagHelperBinding WrapTagHelperBinding(object obj)
+        => Wrap<TagHelperBinding, TagHelperBindingWrapper, IRazorTagHelperBinding>(obj, static obj => new TagHelperBindingWrapper(obj));
+
+    internal static IRazorTagHelperDescriptor WrapTagHelperDescriptor(object obj)
+        => Wrap<TagHelperDescriptor, TagHelperDescriptorWrapper, IRazorTagHelperDescriptor>(obj, static obj => new TagHelperDescriptorWrapper(obj));
+
+    internal static IRazorTagHelperDocumentContext WrapTagHelperDocumentContext(object obj)
+        => Wrap<TagHelperDocumentContext, TagHelperDocumentContextWrapper, IRazorTagHelperDocumentContext>(obj, static obj => new TagHelperDocumentContextWrapper(obj));
+
+    internal static IRazorTagHelperCompletionService WrapTagHelperCompletionService(object obj)
+        => Wrap<TagHelperCompletionService, TagHelperCompletionServiceWrapper, IRazorTagHelperCompletionService>(obj, static obj => new TagHelperCompletionServiceWrapper(obj));
+
+    internal static IRazorTagHelperFactsService WrapTagHelperFactsService(object obj)
+        => Wrap<TagHelperFactsService, TagHelperFactsServiceWrapper, IRazorTagHelperFactsService>(obj, static obj => new TagHelperFactsServiceWrapper(obj));
+
+    internal static IRazorTagMatchingRuleDescriptor WrapTagMatchingRuleDescriptor(object obj)
+        => Wrap<TagMatchingRuleDescriptor, TagMatchingRuleDescriptorWrapper, IRazorTagMatchingRuleDescriptor>(obj, static obj => new TagMatchingRuleDescriptorWrapper(obj));
+
+    internal static RazorSourceSpan ConvertSourceSpan(object obj)
+    {
+        var span = CastStruct<SourceSpan>(obj);
+
+        return ConvertSourceSpan(span);
+    }
+
+    [return: NotNullIfNotNull(nameof(obj))]
+    internal static RazorSourceChange? ConvertSourceChange(object? obj)
```

Comment:
Ditto object param here.

Reply:
This has to be `object?` to avoid exposing the compiler's `SourceSpan` type to the legacy editor.